### PR TITLE
Pass unique tunnel ID to Sauce Connect to avoid terminating existing connections

### PR DESCRIPTION
**Pull request by bcjordan**
:sob: [Sauce Connect auto-terminates existing tunnels with the same identifier, and having no identifier set counts as an identifier.](https://support.saucelabs.com/customer/portal/articles/2005361-my-tunnel-closed-with-connection-closed-exit-code-8-)

---

🎭 What happens with concurrent `[test ui]` runs on CircleCI before this change:

😄 VM A chugs along, starts its tunnel, starts running UI tests.

😈 VM B spins up its tunnel, it sees:

```
04 Aug 15:51:03 - Starting secure remote tunnel VM...
04 Aug 15:51:03 - Checking overlap for my domain sauce-connect.proxy, other tunnel domain sauce-connect.proxy.
04 Aug 15:51:03 - Shutting down overlapping tunnel 2cfcc13190aa468c86b1f84e8154a411 for domain sauce-connect.proxy.
04 Aug 15:51:07 - Checking overlap for my domain sauce-connect.proxy, other tunnel domain sauce-connect.proxy.
04 Aug 15:51:07 - Shutting down overlapping tunnel a0f0a4192d854348bab1614b908898ad for domain sauce-connect.proxy.
[... runs tests]
```

😡 Then VM A sees:

```
04 Aug 15:51:12 - Warning: Remote tunnel VM has been terminated by user request. Shutting down.
[...test failure output]
```

And the UI tests run for a while failing a bunch because there is no tunnel running.

---

This PR adds a random string as the Sauce Connect `-i` identifier argument.

### CODE REVIEW

File: lib/rake/circle.rake [hunk 1]
```diff
@@ -4,6 +4,10 @@ require 'cdo/git_utils'
 RUN_UI_TESTS_TAG = '[test ui]'
 RUN_ALL_TESTS_TAG = '[test all]'
 
+def random_string_of_length(n)
+  rand(36**n).to_s(36)
+end
```
Comment (Bjvanminnen) [inline lib/rake/circle.rake]: Would it make more sense to use an existing random string generator? Maybe something like SecureRandom.uuid? (not sure what's standard in ruby land - that's just something I found). I realize we don't really care about it being truly random, so maybe what you have is fine.

**Comment by Bjvanminnen** [comment]
lgtm

File: lib/rake/circle.rake (hunk 1 above)
Comment (bcjordan) [inline lib/rake/circle.rake]: Good call. Wasn't sure if the `-i` argument had requirements around non-alphanumeric key elements, but testing now looks like it handles SecureRandom.uuid's `-`s no problem.

**Comment by bcjordan** [comment]
Worth noting the first Circle run above had a failure that appears like the tunnel was not operational the entire run.

**Comment by bcjordan** [comment]
@Bjvanminnen fixed issue—have to pass tunnel identifier through when connecting to Sauce Labs. PTAL (merging to unblock other CircleCI improvement branches)

**Pull request merged**